src/mo_frontend/typing.ml

```diff
@@ -2544,6 +2569,39 @@ and infer_obj env s exp_opt dec_fields at : T.typ =
   end;
   t
 
+and validate_parenthetical env typ_opt = function
+  | None -> ()
+  | Some par ->
+     let env = { env with async = C.NullCap } in
+     begin match typ_opt with
+     | Some fun_ty when T.is_func fun_ty ->
+       let s, _, _, _, ts2 = T.as_func fun_ty in
+       begin match ts2 with
+       | _ when T.is_shared_sort s -> ()
+       | [cod] when T.is_async cod -> ()
+       | _ -> warn env par.at "M0210" "unexpected parenthetical note on a non-send call"
+       end
+     | _ -> ()
+     end;
+     let checked = T.[ cycles_fld; timeout_fld ] in
+     let [@warning "-8"] par_infer env { it = ObjE (bases, fields); _ } =
+       infer_check_bases_fields env checked par.at bases fields in
+     let attrs = infer_exp_wrapper par_infer T.as_immut env par in
```

Comment:
The wrapper stuff is a bit ugly, but I guess you need to pass down checked to the infer_check_bases_fields. Hmm, ok.

I do wonder if it might just be simpler for the parser to insert a default base with zero cycles and timeout, and just check that the annotation is a subtype of {cycles : Nat; timeout:Nat32} and then warn about any additional field. But maybe that's not good enough to resolve the overloading of a timeout literal...(which would default to Nat). I can live with this...

Reply:
`(with timeout = 0)` has a very different meaning than not specifying a `timeout`. The former is a best-effort call, the latter and unbounded-wait one.

Reply:
Ok, but you could use `timeout = ?n` instead, which would be nice if you want to explicitly set no timeout.

However, a quick experiment in the playground reveals that we don't currently implement checking mode for `object combinations`, so one would need to extend checking mode to pass down any expected field types to the field extensions for this to work with timout : (?)Nat32. (since inference defaults to Nat). I guess you would be stuck doing inference on the bases if there is more than one though... For  single or no base, I think you could just remove the overridden fields from the expected type.

Reply:
Yeah one could do that, at the expense of complicating things. The current way is "timeout is mentioned in the parenthetical" then best effort, otherwise unbounded wait. The feature is marked as experimental, so we can change it if users really want a change.

Reply:
Also users could have have `(waiting with cycles = 333)` where `waiting` is just an empty record.